rpc/src/v1/impls/devel.rs

```diff
@@ -93,4 +99,215 @@ where
     fn get_block_sync_peers(&self) -> Result<Vec<SocketAddr>> {
         Ok(self.block_sync.get_peers().into_iter().map(|node_id| node_id.into_addr().into()).collect())
     }
+
+    fn test_tps(&self, setting: TPSTestSetting) -> Result<f64> {
+        let fee = self.client.common_params().min_parcel_cost;
+        let shard_id = 0;
+        let network_id = self.client.common_params().network_id;
+        let lock_script_hash = H160::from("b042ad154a3359d276835c903587ebafefea22af");
+        let approver = None;
+
+        // NOTE: Supposing solo network
+        let genesis_secret: Private = "ede1d4ccb4ec9a8bbbae9a13db3f4a7b56ea04189be86ac3a6a439d9a0a1addd".into();
+        let genesis_keypair = KeyPair::from_private(genesis_secret).map_err(errors::parcel_core)?;
+        let base_seq = self.client.seq(&genesis_keypair.address(), BlockId::Latest).unwrap();
+
+        match setting {
+            TPSTestSetting::PaymentOnly(count) => {
+                let start_time = PreciseTime::now();
+                for i in 0..count {
+                    let address = Address::random();
+
+                    let parcel = Parcel {
+                        seq: base_seq + i,
+                        fee,
+                        network_id,
+                        action: Action::Payment {
+                            receiver: address,
+                            amount: 1,
+                        },
+                    };
+                    let signed = SignedParcel::new_with_sign(parcel, genesis_keypair.private());
+                    let signed_hash = signed.hash();
+                    self.miner.import_own_parcel(&*self.client, signed).map_err(errors::parcel_core)?;
+
+                    if i == count - 1 {
+                        while self.client.parcel_invoice(&signed_hash.into()).is_none() {}
```

Comment:
What happens if we manually call `update_sealing` before waiting the parcel invoice?

Reply:
I think it does not do anything by `reseal_min_period`. What's the reason to put `update_sealing`?

Reply:
Oh, it updates `next_allowed_reseal` but does not check it. I think it's possible to do so.

Reply:
Added